core/designsystem/src/main/java/co/kr/tnt/designsystem/component/TextField.kt

```diff
@@ -0,0 +1,240 @@
+                    textStyle = TextStyle(color = TnTTheme.colors.neutralColors.Neutral600),
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .onFocusChanged { focusState ->
+                            isFocused = focusState.isFocused
+                        }
+                        .padding(vertical = 8.dp),
+                    decorationBox = { innerTextField ->
+                        innerTextField()
+                    },
+                )
+            }
+
+            Box(
+                modifier = Modifier
+                    .wrapContentSize(Alignment.Center)
+                    .align(Alignment.CenterVertically),
+            ) {
+                Column(
+                    modifier = Modifier.padding(vertical = 4.dp),
+                ) {
+                    rightComponent()
+                }
+            }
+        }
+
+        HorizontalDivider(
+            thickness = 1.dp,
+            color = lineColor,
+        )
+
+        if (showWarning && !warningMessage.isNullOrEmpty()) {
+            Text(
+                text = warningMessage,
+                style = TnTTheme.typography.body2Medium,
+                color = TnTTheme.colors.mainColors.Red500,
+                modifier = Modifier.padding(top = 6.dp),
+            )
+        }
+    }
+}
+
+@Composable
+fun TnTLabeledTextField(
+    title: String,
+    value: String,
+    placeholder: String,
+    onValueChange: (String) -> Unit,
+    maxLength: Int = 15,
+    isSingleLine: Boolean = false,
+    showWarning: Boolean = false,
+    optional: Boolean = false,
+    warningMessage: String? = null,
+    rightComponent: @Composable () -> Unit = {},
+    modifier: Modifier = Modifier,
+) {
+    val counterColor = when (showWarning) {
+        true -> TnTTheme.colors.mainColors.Red500
+        false -> TnTTheme.colors.neutralColors.Neutral400
+    }
+
+    Column(modifier = modifier.fillMaxWidth()) {
+        Row(
+            horizontalArrangement = Arrangement.SpaceBetween,
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier
+                .padding(bottom = 7.dp, end = 4.dp)
+                .fillMaxWidth(),
+        ) {
+            Row {
+                Text(
+                    text = title,
+                    style = TnTTheme.typography.body1Bold,
+                    color = TnTTheme.colors.neutralColors.Neutral900,
+                )
+                if (!optional) {
+                    Text(
+                        text = "*",
+                        style = TnTTheme.typography.body1Bold,
+                        color = TnTTheme.colors.mainColors.Red500,
+                    )
+                }
+            }
+
+            Text(
+                text = "${value.length}/$maxLength",
+                style = TnTTheme.typography.label1Medium,
+                color = counterColor,
+            )
```

Comment:
```suggestion
            Text(
                text = title,
                style = TnTTheme.typography.body1Bold,
                color = TnTTheme.colors.neutralColors.Neutral900,
            )
            if (!optional) {
                Text(
                    text = "*",
                    style = TnTTheme.typography.body1Bold,
                    color = TnTTheme.colors.mainColors.Red500,
                )
            }
            Spacer(Modifier.weight(1f))
            Text(
                text = "${value.length}/$maxLength",
                style = TnTTheme.typography.label1Medium,
                color = counterColor,
            )
```

끝으로 제한 글자 수 위치시키기 위함이라면, 위와 같이 `Spacer` 를 활용할 수 있을 것 같아요!

Reply:
옹 `Spacer`가 있었네요 수정했습니다!